extensions/wikia/ArticleComments/classes/ArticleComment.class.php

```diff
@@ -1588,4 +1522,93 @@ static private function isBadAccessError( $errors ) {
 		return in_array( 'badaccess-groups', $errors ) || in_array( 'badaccess-group0', $errors );
 	}
 
+	/**
+	 * Manages permissions related to article and blog comments
+	 * Hook: userCan
+	 *
+	 * @param Title $title
+	 * @param User $user
+	 * @param string $action
+	 * @param bool $result Whether $user can perform $action on $title
+	 * @return bool Whether to continue checking hooks
+	 */
+	static public function userCan( Title &$title, User &$user, $action, &$result ) {
+		$app = F::app();
+		$commentsNS = $app->wg->ArticleCommentsNamespaces;
+		$ns = $title->getNamespace();
+
+		// Only handle article and blog comments
+		if ( !in_array( MWNamespace::getSubject( $ns ), $commentsNS ) ||
+			!ArticleComment::isTitleComment( $title ) ) {
+			return true;
+		}
+
+		$comment = ArticleComment::newFromTitle( $title );
+		$isBlog = ( $app->wg->EnableBlogArticles && ArticleComment::isBlog( $title ) );
+
+		$return = true;
+		switch ( $action ) {
+			// Creating article comments requires 'commentcreate' permission
+			// For blogs, additionally check if the owner has enabled commenting+
+			case 'create':
+				// We have to check these permissions on the parent article
+				// due to the chicken-and-egg problem inherent in the design
+				$result = self::userCanCommentOn( $comment->getArticleTitle(), $user );
+				$return = false;
```

Comment:
basically, `$result` specifies if this handler thinks this action should be allowed. If `$return` is true, any other handlers executed after this one might modify `$result`. If it's false, no further handlers are executed, so `$result` is definitely the value specified in this handler.